ml-image/scripts/evaluation_annotation.py

```diff
@@ -93,24 +98,143 @@ def label_results(detections: list, subframes: list, subframe_indices: list) ->
+            frame_annotations[matches[j]]['predicted_ethnicity'] = ethnicities[j]
+            frame_annotations[matches[j]]['predicted_ethnicity_confs'] = ethnicity_confs[j]        
+
+        frame_scores.extend(score_evaluation(frame_annotations, evaluation_type))
+    return np.mean(frame_scores)
+
+
+def score_evaluation(frame_annotations:dict, evaluation_type:str = 'binary') -> list:
+    age_labels = {
+        0 : '0-2', 1 : '3-9', 2 : '10-19', 3 : '20-29', 4 : '30-39',
+        5 : '40-49', 6 : '50-59', 7 : '60-69', 8 : '70+',
+        }
+
+    gender_labels = {
+        0 : 'Male', 1 : 'Female',
+        }
+
+    ethnicity_labels = {
+        0 : 'White', 1 : 'Black', 2 : 'Latino_Hispanic', 3 : 'East Asian',
+        4 : 'Southeast Asian', 5 : 'Indian', 6 : 'Middle Eastern',
+        }
+    
+    rev_age_labels = {age_labels[key] : key for key in sorted(age_labels)}
+    rev_gender_labels = {gender_labels[key] : key for key in sorted(gender_labels)}
+    rev_ethnicity_labels = {ethnicity_labels[key] : key for key in sorted(ethnicity_labels)}
+                      
+    annotation_scores = []
+    keys = ['age', 'gender', 'ethnicity']
+    feat_dicts = [rev_age_labels, rev_gender_labels, rev_ethnicity_labels]
+    for annotation in frame_annotations.values():
+        score = 0
+        for key, feat_dict in zip(keys, feat_dicts, strict=False) :
+            if annotation[key] == annotation[f'predicted_{key}'] :
+                score += 0
+            else :
+                match evaluation_type :
+                    case 'binary' :
+                        score += 1
+                    case _ :
+                        annot_idx = feat_dict[annotation[key]]
+                        score += 1 - annotation[f'predicted_{key}_confs'][annot_idx]
+                        
+        annotation_scores.append(score / len(keys))
+    return annotation_scores
+
+
+def save_displayed_annotations(trailer_directory:str) -> None:
+    trailer_path = os.path.join('evaluation_trailers', trailer_directory, trailer_directory)
+    os.makedirs(os.path.join('evaluation_trailers', trailer_directory, 'annotated_frames'), exist_ok=True)
+    with open(f'{trailer_path}_annotated.pkl', 'rb') as infile :
+        annotations = pkl.load(infile)
+    infile.close()
+    frames_indices = [int(key.split('_')[1]) for key in annotations]
+    frames = frame_capture(f'{trailer_path}.mp4')
+    for index in frames_indices :
+        frame_annotations = annotations[f'frame_{index}']
+        frame = frames[index]
+        img_frame = Image.fromarray(frame)
+        imgdraw = deepcopy(img_frame)
+        draw = ImageDraw.Draw(imgdraw)
+        font = ImageFont.load_default(size = 16)
+
+        for i in range(len(frame_annotations)) :
+            x1, y1, x2, y2 = frame_annotations[i]['bbox']
+            match frame_annotations[i]["gender"]:
+                case "Male":
+                    color = (0, 0, 255)
```

Comment:
```suggestion
                    color = (255, 0, 0) 
```
cv2 est dans l'espace BGR alors que PIL est dans RGB donc on doit inverser les dimensions si on veut matcher les couleurs de ```utils.py/draw_predictions_on_video```. Ou alors tu peux modifier le code pour faire du cv2